### PR TITLE
Synthetic `_source`: `ignore_malformed` for `ip`

### PR DESCRIPTION
**Pull request by nik9000**
This adds synthetic `_source` support for `ip` fields with `ignore_malfored` set to `true`. We save the field values in hidden stored field, just like we do for `ignore_above` keyword fields. Then we load them at load time.

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/index/mapper/IgnoreMalformedStoredValues.java [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+ * {@code _source}.
+ */
+public abstract class IgnoreMalformedStoredValues {
+    /**
+     * Build a {@link StoredField} for the value on which the parser is
+     * currently positioned.
+     * <p>
+     * We try to use {@link StoredField}'s native types for fields where
+     * possible but we have to preserve more type information than
+     * stored fields support, so we encode steal the {@code byte[]}
```
Comment (romseygeek) [inline server/src/main/java/org/elasticsearch/index/mapper/IgnoreMalformedStoredValues.java:37]: I think there's an extra 'steal' here?

**Review by romseygeek — APPROVED** [review]
This is great, thanks @nik9000

**Review by martijnvg — APPROVED** [review]
LGTM - I like the fallback to stored fields incase of ignore malformed.

**Comment by nik9000** [comment]
run elasticsearch-ci/packaging-tests-windows-sample

**Pull request merged**